Tests fail with pytest 3.5.0 DOCTESTS

After updating from pytest from 3.4.2 to 3.5.0, pytest-sugar's testsuite fails:
```
============================= test session starts ==============================
platform linux -- Python 3.6.5, pytest-3.5.0, py-1.5.3, pluggy-0.6.0
rootdir: /build/pytest-sugar-0.9.1, inifile:
plugins: sugar-0.9.1
collected 27 items

test_sugar.py \x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 11%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 22%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 33%]
\x1b[32m\u2713\x1b[0ms\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 44%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 55%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 66%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 77%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0ms\x1b[32m\u2713\x1b[0m [ 88%]
\x1b[32m\u2713\x1b[0ms\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[31m\u2a2f\x1b[0m\x1b[32m\u2713\x1b[0m [100%]

=================================== FAILURES ===================================
___________________ TestTerminalReporter.test_doctest_lineno ___________________

self = <test_sugar.TestTerminalReporter object at 0x7ffff0a768d0>
testdir = <Testdir local('/build/pytest-of-nixbld/pytest-0/test_doctest_lineno0')>

    def test_doctest_lineno(self, testdir):
        """ Test location reported for doctest-modules """

        testdir.makepyfile(
            """
                def foobar():
                    '''
                    >>> foobar()
                    '''
                    raise NotImplementedError
                """
        )
        result = testdir.runpytest('--force-sugar', '--doctest-module')

        assert result.ret == 1, result.stderr.str()
        result.stdout.fnmatch_lines([
            'UNEXPECTED EXCEPTION: NotImplementedError()',
            '*test_doctest_lineno.py:3: UnexpectedException',
            'Results*:',
>           '*-*test_doctest_lineno.py*:3*',
        ])

/build/pytest-sugar-0.9.1/test_sugar.py:514:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_pytest.pytester.LineMatcher object at 0x7ffff0ccb3c8>
lines2 = ['UNEXPECTED EXCEPTION: NotImplementedError()', '*test_doctest_lineno.py:3: UnexpectedException', 'Results*:', '*-*test_doctest_lineno.py*:3*']

    def fnmatch_lines(self, lines2):
        """Search captured text for matching lines using ``fnmatch.fnmatch``.

            The argument is a list of lines which have to match and can use glob
            wildcards.  If they do not match a pytest.fail() is called.  The
            matches and non-matches are also printed on stdout.

            """
>       self._match_lines(lines2, fnmatch, 'fnmatch')
E       Failed: nomatch: 'UNEXPECTED EXCEPTION: NotImplementedError()'
E           and: 'Test session starts (platform: linux, Python 3.6.5, pytest 3.5.0, pytest-sugar 0.9.1)'
E           and: 'rootdir: /build/pytest-of-nixbld/pytest-0/test_doctest_lineno0, inifile:'
E           and: 'plugins: sugar-0.9.1'
E           and: ''
E           and: ''
E           and: '\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015 [doctest] test_doctest_lineno.foobar \u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015'
E           and: '002 '
E           and: '003                 >>> foobar()'
E       exact match: 'UNEXPECTED EXCEPTION: NotImplementedError()'
E       nomatch: '*test_doctest_lineno.py:3: UnexpectedException'
E           and: 'Traceback (most recent call last):'
E           and: ''
E           and: '  File "/nix/store/33v7rg3qmwfj3smvhna5634n7x286wl6-python3-3.6.5/lib/python3.6/doctest.py", line 1330, in __run'
E           and: '    compileflags, 1), test.globs)'
E           and: ''
E           and: '  File "<doctest test_doctest_lineno.foobar[0]>", line 1, in <module>'
E           and: ''
E           and: '  File "/build/pytest-of-nixbld/pytest-0/test_doctest_lineno0/test_doctest_lineno.py", line 5, in foobar'
E           and: '    raise NotImplementedError'
E           and: ''
E           and: 'NotImplementedError'
E           and: ''
E       fnmatch: '*test_doctest_lineno.py:3: UnexpectedException'
E          with: '/build/pytest-of-nixbld/pytest-0/test_doctest_lineno0/test_doctest_lineno.py:3: UnexpectedException'
E       nomatch: 'Results*:'
E           and: '\r'
E           and: '\\r \\x1b[36m\\x1b[0mtest_doctest_lineno.py\\x1b[0m \\x1b[31m\u2a2f\\x1b[0m                                        \\x1b[31m100% \\x1b[0m\\x1b[40m\\x1b[31m\u2588\\x1b[0m\\x1b[40m\\x1b[31m\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\\x1b[0m'
E           and: ''
E       fnmatch: 'Results*:'
E          with: 'Results (0.01s):'
E       nomatch: '*-*test_doctest_lineno.py*:3*'
E           and: '\x1b[31m       1 failed\x1b[0m'
E           and: '         - \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m:2 \x1b[31m[doctest] test_doctest_lineno.foobar\x1b[0m'
E           and: ''
E       remains unmatched: '*-*test_doctest_lineno.py*:3*'

/nix/store/chvgi987hbqcw7zx0q8qzbk4d1dg9mnd-python3.6-pytest-3.5.0/lib/python3.6/site-packages/_pytest/pytester.py:1175: Failed
----------------------------- Captured stdout call -----------------------------
Test session starts (platform: linux, Python 3.6.5, pytest 3.5.0, pytest-sugar 0.9.1)\nrootdir: /build/pytest-of-nixbld/pytest-0/test_doctest_lineno0, inifile:\nplugins: sugar-0.9.1\n\n\n\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015 [doctest] test_doctest_lineno.foobar \u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\n002 \n003                 >>> foobar()\nUNEXPECTED EXCEPTION: NotImplementedError()\nTraceback (most recent call last):\n\n  File "/nix/store/33v7rg3qmwfj3smvhna5634n7x286wl6-python3-3.6.5/lib/python3.6/doctest.py", line 1330, in __run\n    compileflags, 1), test.globs)\n\n  File "<doctest test_doctest_lineno.foobar[0]>", line 1, in <module>\n\n  File "/build/pytest-of-nixbld/pytest-0/test_doctest_lineno0/test_doctest_lineno.py", line 5, in foobar\n    raise NotImplementedError\n\nNotImplementedError\n\n/build/pytest-of-nixbld/pytest-0/test_doctest_lineno0/test_doctest_lineno.py:3: UnexpectedException\n\r\n\r \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m \x1b[31m\u2a2f\x1b[0m                                        \x1b[31m100% \x1b[0m\x1b[40m\x1b[31m\u2588\x1b[0m\x1b[40m\x1b[31m\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\x1b[0m\n\nResults (0.01s):\n\x1b[31m       1 failed\x1b[0m\n         - \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m:2 \x1b[31m[doctest] test_doctest_lineno.foobar\x1b[0m
================ 1 failed, 76 passed, 3 skipped in 2.16 seconds ================
```

--- a/pytest_sugar.py
+++ b/pytest_sugar.py
@@ -64,8 +64,8 @@ PROGRESS_BAR_BLOCKS = [
 ]
 
 
-def flatten(l):
-    for x in l:
+def flatten(seq):
+    for x in seq:
         if isinstance(x, (list, tuple)):
             for y in flatten(x):
                 yield y
@@ -589,8 +589,8 @@ class SugarTerminalReporter(TerminalReporter):
         return crashline
 
     def _get_lineno_from_report(self, report):
-        # Doctest failure reports changed the attribute where longrepr were stored in pytest>3.10
-        # to reprlocation_lines, a list of (ReprFileLocation, lines)
+        # Doctest failures in pytest>3.10 are stored in
+        # reprlocation_lines, a list of (ReprFileLocation, lines)
         try:
             location, lines = report.longrepr.reprlocation_lines[0]
             return location.lineno

--- a/pytest_sugar.py
+++ b/pytest_sugar.py
@@ -238,6 +238,7 @@ class SugarTerminalReporter(TerminalReporter):
             self.print_failure(report)
 
     def pytest_sessionstart(self, session):
+        self._session = session
         self._sessionstarttime = py.std.time.time()
         verinfo = ".".join(map(str, sys.version_info[:3]))
         self.write_line(

--- a/pytest_sugar.py
+++ b/pytest_sugar.py
@@ -531,14 +531,7 @@ class SugarTerminalReporter(TerminalReporter):
                 else:
                     path = os.path.dirname(report.location[0])
                     name = os.path.basename(report.location[0])
-                    # Doctest failure reports have lineno=None at least up to
-                    # pytest==3.0.7, but it is available via longrepr object.
-                    try:
-                        lineno = report.longrepr.reprlocation.lineno
-                    except AttributeError:
-                        lineno = report.location[1]
-                        if lineno is not None:
-                            lineno += 1
+                    lineno = self._get_lineno_from_report(report)
                     crashline = '%s%s%s:%s %s' % (
                         colored(path, THEME['path']),
                         '/' if path else '',
@@ -594,6 +587,24 @@ class SugarTerminalReporter(TerminalReporter):
                 crashline = crashline.decode(encoding, errors='replace')
 
         return crashline
+
+    def _get_lineno_from_report(self, report):
+        # Doctest failure reports changed the attribute where longrepr were stored in pytest>3.10
+        # to reprlocation_lines, a list of (ReprFileLocation, lines)
+        try:
+            location, lines = report.longrepr.reprlocation_lines[0]
+            return location.lineno
+        except AttributeError:
+            pass
+        # Doctest failure reports have lineno=None at least up to
+        # pytest==3.0.7, but it is available via longrepr object.
+        try:
+            return report.longrepr.reprlocation.lineno
+        except AttributeError:
+            lineno = report.location[1]
+            if lineno is not None:
+                lineno += 1
+            return lineno
 
     def summary_failures(self):
         # Prevent failure summary from being shown since we already

--- a/test_sugar.py
+++ b/test_sugar.py
@@ -551,10 +551,6 @@ class TestTerminalReporter(object):
 
         assert result.ret == 1, result.stderr.str()
 
-    @pytest.mark.skipif(
-      LooseVersion(pytest.__version__) >= LooseVersion('3.5'),
-      reason='Temporarily skipping until #134'
-    )
     def test_doctest_lineno(self, testdir):
         """ Test location reported for doctest-modules """
 

--- a/test_sugar.py
+++ b/test_sugar.py
@@ -1,7 +1,6 @@
 # -*- coding: utf-8 -*-
 import pytest
 import re
-from distutils.version import LooseVersion
 from pytest_sugar import strip_colors
 
 pytest_plugins = "pytester"

